spec/formatting.md

```diff
@@ -217,8 +217,8 @@ the following steps are taken:
      bind the _identifier_ of the _option_ to the resolved value in the mapping.
    - Otherwise, bind the _identifier_ of the _option_ to an unresolved value in the mapping.
      (Note that an Unresolved Variable error will have been emitted.)
-4. Remove from the resolved mapping of _options_ any binding for which the value is an unresolved value.
```

Comment:
This seems reasonable, but perhaps we should explicitly allow implementations to remove the value later (to prevent the local equivalent of NPE):

```suggestion
      Implementations MAY later remove this value before calling the _function_.
```

Reply:
I like this suggestion, thanks @aphillips.

I think delegating the decision about whether to remove or not to implementations is the right choice. In some implementations it's going to be natural to use a sentinel value to represent unresolved option values; other implementations may not have such possibility and they can instead choose to completely remove unresolved options from the map before it's passed to the function.

Reply:
I wasn't able to apply this suggestion via GitHub's UI. I pushed 006e60df939c9f5c92426e2a19f6d13b2847d2cd instead.